use a new library to parse command line args

notrodes opened the pull request:
also some minor cleaning

Commit:
use a new library to parse command line args

psvenk commented on serve.js:
```diff
@@ -16,26 +17,18 @@ const args = require('minimist')(process.argv.slice(2));
-app.use(compression({ filter: (..._) => true }));
-app.listen(port, () => console.log(`Aspine listening on port ${port}!`));
+app.use(compression());
+app.listen(program.port, () => console.log(`Aspine listening on port ${program.port}!`));
```
@notrodes Why did you remove the custom `filter` function? I had included it because `compression` by default does not compress all traffic and so feeding it a function that always returns `true` causes all traffic to be compressed.

psvenk replied:
By the way, I think it should be fine to change the function from `(..._) => true` to the more concise `() => true`, but we should double-check that everything is being compressed.

notrodes replied:
i checked and the default compresses based on what data type it is being used. I though that would be better than everything being compressed.

psvenk replied:
I guess the default algorithm changed since I did this in #125, because now the difference in amount of data transferred is negligible between full compression and default compression. In that case, `compression()` is indeed probably better.

psvenk commented on serve.js:
```diff
@@ -47,22 +40,19 @@ if(!(args.hasOwnProperty('insecure') || args._.includes("insecure"))) {
-    }
+    }); // at top of routing calls
+
+    http.createServer(app).listen(8090);
```
```suggestion
```
I think this line can be removed because we do not serve HTTP on production.

psvenk commented on serve.js:
```diff
@@ -47,22 +40,19 @@ if(!(args.hasOwnProperty('insecure') || args._.includes("insecure"))) {
+    http.createServer(app).listen(8090);
+    https.createServer(credentials, app).listen(4430, () =>//443
+        console.log('HTTPS Server running on port 4430') //443
+    );
```
Remove outdated comments
```suggestion
    https.createServer(credentials, app).listen(4430, () =>
        console.log('HTTPS Server running on port 4430')
    );
```

psvenk commented on serve.js:
```diff
@@ -47,22 +40,19 @@ if(!(args.hasOwnProperty('insecure') || args._.includes("insecure"))) {
         }
         // handle port numbers if you need non defaults
-        // res.redirect('https://' + req.host + req.url); // express 3.x
         res.redirect('https://' + req.hostname + req.url); // express 4.x
```
Remove outdated/unnecessary comment
```suggestion
        res.redirect('https://' + req.hostname + req.url);
```

psvenk reviewed (changes requested):
Additional changes which I cannot put as inline comments (because GitHub won't let me comment on lines of code that were not changed):
- `npm install commander` and commit `package.json` and `package-lock.json`.
- `minimist` is no longer needed, so the `require('minimist')` can be removed and we can also `npm uninstall minimist`.
- The variables `privateKey`, `certificate`, and `ca` are not needed and can be folded into the object literal for `credentials`.
- The version number should come dynamically from `package.json`.

psvenk commented on serve.js:
```diff
@@ -16,26 +17,18 @@ const args = require('minimist')(process.argv.slice(2));
+app.listen(program.port, () => console.log(`Aspine listening on port ${program.port}!`));
 
-if(!(args.hasOwnProperty('insecure') || args._.includes("insecure"))) {
+if(!program.insecure) {
```
Fix spacing
```suggestion
if (!program.insecure) {
```

psvenk commented on serve.js:
```diff
@@ -16,26 +17,18 @@ const args = require('minimist')(process.argv.slice(2));
+app.listen(program.port, () => console.log(`Aspine listening on port ${program.port}!`));
 
-if(!(args.hasOwnProperty('insecure') || args._.includes("insecure"))) {
+if(!program.insecure) {
```
This breaks the usage of `npm start insecure`, meaning that we need to update documentation to use `npm start -- -i` instead (which IMO is not very elegant).

notrodes replied:
no, -i or --insecure, which is normal

psvenk replied:
My main concern is not inelegance but that documentation and start scripts need to be updated to use `npm start -- -i`.

notrodes committed:
remove redundant code and comments

Commit:
add and remove deps as appropriate

psvenk committed:
Update documentation for new CLI

Commit:
get version num from package.json

psvenk committed:
More refactoring; make HTTPS port configurable

psvenk committed:
Get version number from git

This is consistent with how the /version endpoint works and also
differentiates between release and development versions.
